app/src/main/java/com/onboarding/pronosticAR/activity/MainActivity.kt

```diff
@@ -16,6 +17,7 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        presenter = MainPresenter(model = MainModel(), view = MainView(this))
+        presenter = MainPresenter(model = MainModel(WeatherService()), view = MainView(this))
+        presenter.getWeatherInfo()
```

Comment:
please use init { ... }  in the presenter if this is the first action after create the class
https://blog.mindorks.com/understanding-init-block-in-kotlin

Reply:
done